lib/renaissance_web/controllers/auction_controller.ex

```diff
@@ -0,0 +1,33 @@
+defmodule RenaissanceWeb.AuctionController do
+  use RenaissanceWeb, :controller
+  alias Renaissance.{Auction, Auctions}
+  alias RenaissanceWeb.Helpers.{Auth}
+
+  def new(conn, _params) do
+    changeset = Auction.changeset(%Auction{})
+
+    if Auth.signed_in?(conn) do
+      render(conn, "new_auction.html", changeset: changeset)
+    else
+      conn
+      |> redirect(to: Routes.login_path(conn, :login))
+    end
+  end
+
+  def create(conn, params) do
+    changeset =
+      conn
+      |> get_session(:current_user)
```

Comment:
I guess it wasn't clear to me in the original PR, but I would not have expected `session[:current_user]` to be the email address. I would have expected something like `get_session(:current_user_id)`. Just use the primary key, then you also avoid having to look up id from email wherever you want to attach something to the current user.

Reply:
agree